src/main/kotlin/cacophony/automata/minimalization/PartitionRefinement.kt

```diff
@@ -0,0 +1,44 @@
+package cacophony.automata.minimalization
+
+// Represents a set of the partition and allow for quick comparisons
+data class PartitionId(
+    val id: Int,
+)
+
+class PartitionRefinement<E>(
+    baseSet: Collection<E>,
+) {
+    private val elementToPartitionId: MutableMap<E, PartitionId> = HashMap()
+    private val partitionToElements: MutableMap<PartitionId, MutableSet<E>> = HashMap()
+
+    init {
+        baseSet.associateWithTo(elementToPartitionId) { PartitionId(0) }
+        partitionToElements[PartitionId(0)] = baseSet.toMutableSet()
+    }
+
+    // Refines the partition
```

Comment:
Please add better comment about this functionality. And end comments with dot '.'